app/services/check_new_matches_service.rb

```diff
@@ -0,0 +1,32 @@
+class CheckNewMatchesService
+  def initialize(params)
```

Comment:
Nice that you used hash for params (it's more flexible), but you can use keywords parameters that has the same purpose but it can allow defaults as well and you don't have to handle the hash:
```
def initialize(new_target:, current_user:)
...
```

More info: https://robots.thoughtbot.com/ruby-2-keyword-arguments